src/main/java/mServer/crawler/sender/phoenix/tasks/AbstractRestTask.java

```diff
@@ -0,0 +1,80 @@
+package mServer.crawler.sender.phoenix.tasks;
+
+import de.mediathekview.mlib.Config;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import mServer.crawler.sender.MediathekReader;
+import mServer.crawler.sender.orf.CrawlerUrlDTO;
+import mServer.crawler.sender.orf.tasks.AbstractUrlTask;
+import org.glassfish.jersey.client.filter.EncodingFilter;
+import org.glassfish.jersey.message.DeflateEncoder;
+import org.glassfish.jersey.message.GZipEncoder;
+
+/**
+ * This task is based on {@link AbstractUrlTask} which takes a
+ * {@link ConcurrentLinkedQueue} of {@link D} and loads the URL with REST as
+ * {@link WebTarget}.
+ *
+ * @author Nicklas Wiegandt (Nicklas2751)<br/>
+ * <b>Mail:</b> nicklas@wiegandt.eu<br/>
+ * <b>Jabber:</b> nicklas2751@elaon.de<br/>
+ *
+ * @param <T> The type of objects which will be created from this task.
+ * @param <D> A sub type of {@link CrawlerUrlDTO} which this task will use to
+ * create the result objects.
+ */
+public abstract class AbstractRestTask<T, D extends CrawlerUrlDTO> extends AbstractUrlTask<T, D> {
+
+  private static final long serialVersionUID = 2590729915326002860L;
+  protected static final String ENCODING_GZIP = "gzip";
+  protected static final String HEADER_ACCEPT_ENCODING = "Accept-Encoding";
+  protected static final String HEADER_AUTHORIZATION = "Authorization";
+  protected static final String AUTHORIZATION_BEARER = "Bearer ";
+
+  protected final transient Optional<String> authKey;
+  private final Client client;
+
+  public AbstractRestTask(final MediathekReader aCrawler,
+          final ConcurrentLinkedQueue<D> aUrlToCrawlDTOs, final Optional<String> aAuthKey) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDTOs" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)